sympy/solvers/solveset.py

```diff
@@ -761,11 +820,33 @@ def _solveset(f, symbol, domain, _check=False):
 
     # assign the solvers to use
     solver = lambda f, x, domain=domain: _solveset(f, x, domain)
-    if domain.is_subset(S.Reals):
-        inverter_func = invert_real
-    else:
-        inverter_func = invert_complex
-    inverter = lambda f, rhs, symbol: inverter_func(f, rhs, symbol, domain)
+
+    # check for Modular equations
+    is_mod = S.false
+    if f.has(Mod):
+        # in case symbol is not defined as integer
+        assumption = symbol.assumptions0
+        assumption['integer'] = True
+        t = Dummy('t', **assumption)
+        f = f.xreplace({symbol: t})
+
+        rep_eq = list(f.atoms(Mod))
+        if all([i.is_integer for i in rep_eq[0].args]):
```

Comment:
I don't understand this check. Can someone explain me the need for this @ishanaj @normalhuman